src/signalrclient/signalr_default_scheduler.cpp

```diff
@@ -0,0 +1,217 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include "stdafx.h"
+#include <assert.h>
+#include "signalr_default_scheduler.h"
+#include <thread>
+
+namespace signalr
+{
+    thread::thread()
+        : m_internals(std::make_shared<internals>())
+    {
+        auto internals = m_internals;
+
+        m_thread = std::thread([=]()
+            {
+                while (true)
+                {
+                    {
+                        signalr_base_cb cb;
+                        {
+                            std::unique_lock<std::mutex> lock(internals->m_callback_lock);
+                            auto& callback = internals->m_callback;
+                            auto& closed = internals->m_closed;
+
+                            if (closed && callback == nullptr)
+                            {
+                                assert(callback == nullptr);
```

Comment:
Why we asserting this? Seems odd to assert something we check right before.

Reply:
Artifact of when I was testing stuff